Vesper-Agents-Suite/loss-reporter/src/agent.ts

```diff
@@ -61,10 +77,31 @@ export const provideHandleTransaction = (web3: Web3): HandleTransaction => {
 
     let findingsWithMetadata: Finding[] = [];
     for (let finding of findings) {
-      const metadataInfo = await vesperFetcher.getMetaData(finding.metadata.strategyAddress, txEvent.blockNumber)
-      const description = finding.description + metadataInfo + ", lossValue = " + finding.metadata.lossValue;
-      const findingWithMetadata = { ...finding, description: description }
-      findingsWithMetadata = findingsWithMetadata.concat(Finding.fromObject(findingWithMetadata));
+      const metadata = await vesperFetcher.getMetaData(finding.metadata.strategyAddress, txEvent.blockNumber)
+      const strategyContract = new web3.eth.Contract(STRATEGY_ABI, finding.metadata.strategyAddress);
+      const poolAddress = await strategyContract.methods.pool().call({});
+      const poolContract = new web3.eth.Contract(POOL_ABI, poolAddress);
+      const collateralToken = await poolContract.methods.token().call({});
+      const strategyDebt = await poolContract.methods.totalDebtOf(finding.metadata.strategyAddress).call({}, txEvent.blockNumber);
+      const strategyLossInTvl = strategyDebt > 0 ? new BigNumber(finding.metadata.lossValue).multipliedBy(100).dividedBy(strategyDebt) : 0
+      const collateralContract = new web3.eth.Contract(POOL_ABI, collateralToken);
+      const decimals = await collateralContract.methods.decimals().call({});
+      const symbol = await collateralContract.methods.symbol().call({});
+
+      const lossInCollateral = new BigNumber(finding.metadata.lossValue).dividedBy(new BigNumber('1e' + decimals.toString()))
+      const { nativeToken, router, usdc } = await getContracts()
+      const routerContract = new web3.eth.Contract(ROUTER_ABI, router);
+      const path = web3.utils.toChecksumAddress(collateralToken) === web3.utils.toChecksumAddress(nativeToken)
+        ? [collateralToken, usdc]
+        : [collateralToken, nativeToken, usdc]
+      const amountsOut = await routerContract.methods.getAmountsOut(finding.metadata.lossValue, path).call({})
+      const lossInUSDC = amountsOut[path.length - 1]
+      const lossInUSD = new BigNumber(lossInUSDC).dividedBy(new BigNumber('1e6'))
+      if (lossInUSD.gt(REPORT_LOSS_LIMIT)) { // raise alert only when loss > REPORT_LOSS_ABOVE USD
+        const description = `${finding.description} ${metadata.strategyName} of ${metadata.poolName}. lossInCollateral = ${lossInCollateral} ${symbol}, lossInUSD = $${lossInUSD.toFixed(2)}, lossOfStrategyTVL = ${strategyLossInTvl.toFixed(2)}%, strategyAddress = ${metadata.strategyAddress}, poolAddress = ${metadata.poolAddress}, network = ${metadata.network}`
```

Comment:
is it possible to send some tests alerts in slack? if yes then I would like this string to be formatted such that it becomes properly formatted when it render in slack. I can work with you on this if you want.

Reply:
In fact, I have looked for it and \n does not work to split into new line.
I will still look for a way to test the formatted messages.

Reply:
@rokso There is a way to format the message for slack.
https://thecodebarbarian.com/working-with-the-slack-api-in-node-js.html
However, The formatting is done by forta code.
We can put these data in `metadata` field but that field is not visible on slack and we have to click on a link to see the metadata.

Looking [at](https://api.slack.com/reference/surfaces/formatting), shows it may work if we use **\n**. I will make the changes and see if this is correctly reflected in alerts.

Reply:
I have deployed with \n changes for eth mainnet. Lets see if we get any new alerts over the weekend.

Reply:
@rokso - \n is working okay now.
https://bloqinc.slack.com/archives/C02RJLCMUM8/p1658104963193559